author: Ed Chin <edchin@google.com>
description: fix build system race

We saw buildd failures but I couldn't reproduce it. Turned out to be a
race; there should be a hard dependency on :proto_ios_shared_v2 by
:proto_v2 , but there's not. Instead, :feed_core_v2 depends upon both
:proto_ios_shared_v2 and :proto_v2, and depending on how many cpu cores or
threads you're building with, maybe you get lucky and :proto_ios_shared_v2
gets built first, or maybe you're unlucky and :proto_v2 gets built first
and then the build fails due to missing (generated) headers.






Fix deps for feed_response.proto

feed_response.proto depends on chrome_feed_response_metadata.proto.

crrev.com/c/4266050 moved chrome_feed_response_metadata.proto to a
separate target, and broke a bot. This fixes that.

Bug: 1422213
Change-Id: I893520234d3624173ea653f2adfb92dfd4735be7
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4328228
Reviewed-by: Dan H <harringtond@chromium.org>
Reviewed-by: Justin DeWitt <dewittj@chromium.org>
Commit-Queue: edchin <edchin@google.com>
Cr-Commit-Position: refs/heads/main@{#1115848}

--- a/components/feed/core/proto/BUILD.gn
+++ b/components/feed/core/proto/BUILD.gn
@@ -66,6 +66,7 @@ proto_library("proto_v2") {
     "v2/wire/xsurface_content.proto",
     "v2/xsurface.proto",
   ]
+  proto_deps = [ ":proto_ios_shared_v2" ]
 }
 
 proto_library("proto_ios_shared_v2") {
